Scrap bugs rarity

erwanlbp opened the pull request:
In response to #6

I did just the bugs for a start, if it's accepted I'll try the other ones.

To scrap the rarity we need to go on every bug page so it takes longer to scrap.

The PR for sure needs a bit of rework, it was to start the thing. So please comment :)

Commit:
#6 Scrap bug rarity

Commit:
oups forgot to remove some comments

SawyerHood commented on scraper/scraper.js:
```diff
@@ -112,8 +113,29 @@ async function loadBugs() {
+    await page.goto(`${RARITY_BASE_URL}${encodeURIComponent(bug.name.replace(' ', '_'))}`);
+    bug.rarity = await page.evaluate(() => {
+      const rarity = document.querySelector("[data-source=rarity] div").textContent;
+      if (rarity.includes("Common")) {
```
I'm fine with keeping these values as either the names they are now or a number value between 1 -> 5. When displaying these we are probably just going to show either the text or if we are fancy a few star icons. I think we should start with just the words

SawyerHood commented:
Thanks for taking this on. We might want to make the scraping a little more efficient. This is going to make it hit 100s of pages (if we enable it for fish as well). If we do this we will probably want to run multiple pages at once (this work can be done in parallel). I think we should rework this to do that before I merge in. Let me know if you need any help!

SawyerHood reviewed (changes requested):
See above ^ I forgot to request changes

Commit:
parallel scrap + no star, just words

Commit:
do not run full parallel, just on multiple workers

erwanlbp commented:
> Thanks for taking this on. We might want to make the scraping a little more efficient. This is going to make it hit 100s of pages (if we enable it for fish as well). If we do this we will probably want to run multiple pages at once (this work can be done in parallel). I think we should rework this to do that before I merge in. Let me know if you need any help!

I changed it to run in parallel, but loading 80+ tabs in parallel take some times :joy: 
So I changed it to run in parallel by chunks.
I timed :
- 45s for 4 parallel chunks
- more than 50s for 2 chunks
- around 52s for 3 chunks
- around 46s for the sequential

My machine is not a beast and I'm not equipped with fiber.

What do you think ?

SawyerHood commented on scraper/scraper.js:
```diff
@@ -112,13 +113,49 @@ async function loadBugs() {
+async function loadBugsRarity(bugs, arrayStart, arrayEnd, browser) {
+  for (let i = arrayStart; i < arrayEnd; i++) {
+    const bug = bugs[i];
+    console.log(`[${arrayStart},${arrayEnd}] [${i}] Scrap bug rarity ${bug.name}`);
```
This should be "scrape" not scrap

SawyerHood reviewed (approved):
Sorry I've been busy with work! This looks like a good starting point. In the future we might want to find a better data source for all of the data (I actually was able to rip all of the assets straight from the game on my computer) so we don't have to wait so long to scrape everything. This looks awesome!
